src/fu-util-common.c

```diff
@@ -393,6 +393,63 @@ fu_util_update_reboot (GError **error)
 	return val != NULL;
 }
 
+gboolean
+fu_util_prompt_confirm (FwupdDevice *device, FwupdRelease *release, GError **error)
+{
+	const gchar *desc_tmp;
+	gint vercmp;
+	g_autoptr(GString) str = g_string_new (NULL);
+
+	vercmp = fu_common_vercmp_full (fwupd_release_get_version (release),
+				        fu_device_get_version (device),
+				        fwupd_device_get_version_format (device));
+	if (vercmp < 0) {
+		/* TRANSLATORS: message letting the user know an downgrade is available
+		 * %1 is the device name and %2 and %3 are version strings */
+		g_string_append_printf (str, _("Downgrade %s from %s to %s:"),
+					fwupd_device_get_name (device),
+					fwupd_device_get_version (device),
+					fwupd_release_get_version (release));
+	} else if (vercmp > 0) {
+		/* TRANSLATORS: message letting the user know an upgrade is available
+		 * %1 is the device name and %2 and %3 are version strings */
+		g_string_append_printf (str, _("Upgrade %s from %s to %s:"),
+					fwupd_device_get_name (device),
+					fwupd_device_get_version (device),
+					fwupd_release_get_version (release));
+	} else {
+		/* TRANSLATORS: message letting the user know an upgrade is available
+		 * %1 is the device name and %2 is a version string */
+		g_string_append_printf (str, _("Reinstall %s to %s:"),
+					fwupd_device_get_name (device),
+					fwupd_release_get_version (release));
+	}
+
+	/* no description provided */
+	desc_tmp = fwupd_release_get_description (release);
+	if (desc_tmp != NULL) {
+		g_autofree gchar *desc = fu_util_convert_description (desc_tmp, NULL);
+		if (desc != NULL)
+			g_string_append_printf (str, "%s", desc);
+		fu_util_warning_box (str->str, 80);
+	}
+
+	/* ask for confirmation */
+	g_print ("\n%s [Y|n]: ",
+		/* TRANSLATORS: prompt to apply the update */
+		_("Perform operation?"));
```

Comment:
Don't we also have a prompt when the device will reboot?  So this will manifest as two prompts?  I think we need to move the reboot message into this function so we only have one prompt.

Reply:
True, willfix.

Reply:
Combining them makes the PR much nicer too... Updated.